orttraining/orttraining/training_api/include/onnxruntime_training_cxx_inline.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "onnxruntime_training_c_api.h"
+#include "onnxruntime_cxx_api.h"
+
+namespace Ort {
+
+inline TrainingSession::TrainingSession(const SessionOptions& session_options,
+                                        CheckpointState& checkpoint_state,
+                                        const std::basic_string<ORTCHAR_T>& train_model_path,
+                                        const std::optional<std::basic_string<ORTCHAR_T>>& eval_model_path,
+                                        const std::optional<std::basic_string<ORTCHAR_T>>& optimizer_model_path) {
+  Env env = Env();
+  ThrowOnError(GetTrainingApi().CreateTrainingSession(
+      env, session_options, checkpoint_state,
+      train_model_path.c_str(),
+      eval_model_path.has_value() ? eval_model_path.value().c_str() : nullptr,
+      optimizer_model_path.has_value() ? optimizer_model_path.value().c_str() : nullptr,
+      &p_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetTrainingModelOutputCount(p_, &training_model_output_count_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetEvalModelOutputCount(p_, &eval_model_output_count_));
+}
+
+inline std::vector<Value> TrainingSession::TrainStep(const std::vector<Value>& input_values) {
+  std::vector<Value> output_values;
+  for (size_t i = 0; i < training_model_output_count_; i++) output_values.emplace_back(nullptr);
+  auto ort_input_values = reinterpret_cast<const OrtValue**>(const_cast<Value*>(input_values.data()));
```

Comment:
Actually, I picked up this part of the code from [`onnxruntime_cxx_inline.h`](https://github.com/microsoft/onnxruntime/blob/c20abcab87360eeea5de72fb951d5466adfacc99/include/onnxruntime/core/session/onnxruntime_cxx_inline.h#L630-L633).

Let me try to see why the `const_cast` was added. If not needed, I will remove it from here.

Reply:
This `const_cast` is needed because the API expects `const std::vector<Value>`. `reinterpret_cast` won't cast away the constness. Either we change the api to accept a non const ref to `input_values` or we keep the `const_cast`. What are your thoughts?

Reply:
I think keeping `input_values` const is good.
hm, I don't see why we need to remove const qualification. could it be something like this?

```c++
  auto ort_input_values = reinterpret_cast<const OrtValue* const *>(input_values.data());
```

Reply:
That worked. Thanks for that comment.